app-router/installation.mdx

```diff
@@ -0,0 +1,168 @@
+    using [Next.js 13.4](https://nextjs.org/blog/next-12) or a later version that is using app router.
+
+    <Info>
+      If you are not using app router, here's how to [incrementally adopt](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration) it.
+    </Info>
+
+  </Step>
+  <Step title="Install dependencies">
+  Install the `@makeswift/runtime` package. This package contains all of the necessary
+  code to integrate Makeswift into your Next.js app.
+
+```bash
+npm install @makeswift/runtime
+```
+
+  </Step>
+  <Step title="Add API key to environment variables">
+  
+Requesting data through the `Makeswift` client requires a site API key from Makeswift. In the Makeswift builder, go to **Settings > Host** and copy the API key for the site.
+
+<Frame>
+  <img src="/images/site-api-key.gif" alt="How to get the site API key" />
+</Frame>
+
+Once the API key is in your clipboard, open your [`.env.local`](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables) file and paste the snippet below.
+
+```sh
+MAKESWIFT_SITE_API_KEY=paste-your-api-key-here
+```
+
+  </Step>
+  <Step title="Add Makeswift runtime and client">
+  
+Create files for the Makeswift [runtime](/reference/runtime/constructor) and [client](/reference/client/constructor).
+
+<CodeGroup>
+  <RuntimeExample />
+  <ClientExample />
+</CodeGroup>
+
+</Step>
+  <Step title="Add the Makeswift API handler">
+  
+Similar to [NextAuth.js](https://next-auth.js.org/), Makeswift uses an API handler to communicate with your Next.js app. Create the file `app/api/makeswift/[...makeswift]/route.ts`. It's important that this directory and file structure is followed precisely.
+
+<ApiHandlerExample />
+
+This API route adds support for
+[preview Mode](https://nextjs.org/docs/pages/building-your-application/configuring/preview-mode),
+[on-demand revalidation](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation),
+and other features that make Makeswift work seamlessly with your Next.js app.
+
+  </Step>
+  <Step title="Add the Next.js plugin">
+  
+Next.js plugins are configured in the project's next.config.js file by wrapping `nextConfig`. The Makeswift Next.js plugin whitelists Makeswift image domains and sets up rewrites to enable preview mode in the Makeswift builder.
+
+<NextConfigExample />
+
+  </Step>
+  <Step title="Create Makeswift provider component">
+    Create a client component for the Makeswift providers.
+
+    <ProvidersExample />
+
+  </Step>
+  <Step title="Update the root layout">
+    Wrap your app with the `MakeswiftProviders` component created in the last step and add the `DraftModeScript` to the `head` of your root layout. This script enables draft mode when inside the Makeswift builder.
+
+    <LayoutExample />
+
+  </Step>
+  <Step title="Register components with Makeswift">
```

Comment:
Something we don't want to missed about component registration is that you have to import the registration in both the root layout and the `providers.tsx`

Reply:
I might reverse the order here so you can do the one two three

1. register some components
2. add a providers file with component registration import included
3. update root layout with the component registration import included

Reply:
✅